File: src/IsbInterop/Tasks/Proxies/Notice.vb
﻿Imports IsbInterop.Base.Proxies

Namespace Tasks.Proxies

  ''' <summary>
  ''' Обертка над INotice.
  ''' </summary>
  Friend Class Notice
    Inherits CustomJob(Of INoticeInfo)
    Implements INotice

    ''' <summary>
    ''' Получает информацию об объекте.
    ''' </summary>
    ''' <returns>Информация об объекте.</returns>
    Public Overrides Function GetInfo() As INoticeInfo
      Dim rcwINoticeInfo = GetRcwObjectInfo()
      Return New NoticeInfo(rcwINoticeInfo, Scope)
    End Function

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwINotice">СOM-объект INotice.</param>
    ''' <param name="scope">Область видимости.</param>
    Public Sub New(rcwINotice As Object, scope As IScope)
      MyBase.New(rcwINotice, scope)
    End Sub
  End Class
End Namespace

File: src/IsbInterop/Tasks/Proxies/Notice.cs
using IsbInterop.Base.Proxies;
using IsbInterop.Edms.Proxies;

namespace IsbInterop.Tasks.Proxies
{
  /// <summary>
  /// Обертка над INotice.
  /// </summary>
  internal class Notice : CustomJob<INoticeInfo>, INotice
  {
    /// <summary>
    /// Получает информацию об объекте.
    /// </summary>
    /// <returns>Информация об объекте.</returns>
    public override INoticeInfo GetInfo()
    {
      var rcwINoticeInfo = GetRcwObjectInfo();
      return new NoticeInfo(rcwINoticeInfo, Scope);
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwINotice">СOM-объект INotice.</param>
    /// <param name="scope">Область видимости.</param>
    public Notice(object rcwINotice, IScope scope) : base(rcwINotice, scope) { }
  }
}
